### DISCUSSION TITLE
Evaluation metrics for T5?

### DISCUSSION THREAD

**Discussion by KaiserWhoLearns** [allenai/allennlp#5325] [Q&A]
For the[ code piece of T5](https://github.com/allenai/allennlp-models/blob/main/allennlp_models/generation/models/t5.py) I am looking at, it seems like the only evaluation metrics are set to be ROUGE and BLUE, which seems to belong to summarization tasks. What I am currently doing is to copy the file and edit it on my end. I am pretty new to allennlp so just in case that there may be some tricks without changing much code. What may be the most convenient way to customize the metrics?

**Comment by epwalsh** [comment]
Yea, there's no good reason those should be hardcoded. We should do something similar to what we've done in CopyNet: https://github.com/allenai/allennlp-models/blob/db0e21ae116e5d241252e4482bb8b9514645b979/allennlp_models/generation/models/copynet_seq2seq.py#L88-L89

Would you like to make a PR?
